frontend/web/components/modals/CreateSegment.tsx

```diff
@@ -408,6 +413,32 @@ const CreateSegment: FC<CreateSegmentType> = ({
               : 'Show condition descriptions'}
           </span>
         </Row>
+        {metadataEnable && (
```

Comment:
I think this whole thing can live inside metadataselect if I'm not mistaken? That way we don't need all the show logic inside createflag/createsegment etc?

Reply:
I'll check it, thanks for the suggestion

Reply:
New component Added